cfme/tests/physical_infrastructure/test_redfish_physical_server_buttons.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+import pytest
+from cfme.utils.appliance.implementations.ui import navigate_to
+from cfme.common.physical_server_views import (
+    PhysicalServerDetailsView,
+)
+from cfme.physical.provider.redfish import RedfishProvider
+
+pytestmark = [pytest.mark.provider([RedfishProvider], scope="function")]
+
+
+@pytest.fixture(scope="function")
+def physical_server(appliance, provider, setup_provider_funcscope):
+    """Get and return the first physical server."""
+    yield appliance.collections.redfish_physical_servers.all(provider)[0]
+
+
+def assert_message(physical_server, provider, expected_message):
+    """
+    Assert that the physical server details view displays the requested message.
+
+    Args:
+      physical_server: check the details view of this physical server.
+      provider: the RedfishProvider instance
+      expected_message: the message we expect to be displayed in the flash alert
+    """
+    view = provider.create_view(PhysicalServerDetailsView, physical_server)
+    view.flash.assert_message(expected_message)
+
+
+# Power Button
+def test_redfish_server_power_off(physical_server, provider):
+    physical_server.power_off()
```

Comment:
What if server is powered off already? I mean, you pick first physical server from VMDB so you cannot know, right? I assume the "Power Off" button will be grayed-out in this case which will result in test failure. Similar goes for other tests.

Reply:
All the buttons are always available, so no option is greyed-out as a function of the server's known state.